### PR TITLE
feat: 5.x Rename extension vars, provide state_id to sub-module, Multus

### PR DESCRIPTION
**Pull request by devoncrouse**
- Add `state_id`, `http` provider for extensions, update locals for manifest paths
- Rename extension variables to fit convention: `..._enabled` -> `..._install`
- Add Multus extension

### CODE REVIEW

File: variables-extensions.tf [hunk 1]
```diff
@@ -95,7 +121,7 @@ variable "metrics_server_helm_values_files" {
 
 # Cluster autoscaler
 
-variable "cluster_autoscaler_enabled" {
+variable "cluster_autoscaler_install" {
```
Comment (hyder) [inline variables-extensions.tf:124]: I thought we supported autoscaler?

File: variables-extensions.tf [hunk 2]
```diff
@@ -61,16 +61,42 @@ variable "calico_staging_dir" {
   type        = string
 }
 
+# CNI: Multus
+
+variable "multus_install" {
```
Comment (hyder) [inline variables-extensions.tf:66]: I assume we can do both and not have to pick between the 2?

File: variables-extensions.tf (hunk 1 above)
Comment (devoncrouse) [reply variables-extensions.tf:124]: Yeah, this is just fixing the names you brought up in an earlier PR.

File: variables-extensions.tf (hunk 2 above)
Comment (devoncrouse) [reply variables-extensions.tf:66]: Yes that's right.

**Review by hyder — APPROVED** [review]
Approved

**Pull request merged**